src/toil/jobStores/azureJobStore.py

```diff
@@ -278,9 +278,11 @@ def getSize(cls, url):
     @classmethod
     def _readFromUrl(cls, url, writable):
         blob = cls._parseWasbUrl(url)
-        blob.service.get_blob_to_file(container_name=blob.container,
-                                      blob_name=blob.name,
-                                      stream=writable)
+        for attempt in retry_azure():
+            with attempt:
+                blob.service.get_blob_to_file(container_name=blob.container,
```

Comment:
There's no good solution to this, but just want to point out that this might rarely end up writing a corrupted file with the same part represented two or more times, if you get something like ECONNRESET in the middle of a stream. I can't think of any good way around that, so I think sticking with this behavior is OK. AWSJobStore seems to do something very similar in `FileInfo.downloadStream`, and I assume it hasn't really caused problems, so it's probably only a theoretical problem.

Otherwise, LGTM!